gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/PlanServiceImpl.java

```diff
@@ -610,4 +650,19 @@ private DefinitionVersion getApiDefinitionVersion(Api api) {
             return V2;
         }
     }
+
+    private void validateTags(Set<String> tags, Api api) {
```

Comment:
You should factorize this code with V4 and use an appropriate service see Validation should be made using a dedicated service as for any others attributes see io.gravitee.rest.api.service.v4.impl.validation.ApiValidationServiceImpl